### PR TITLE
feat: Add support for headerVariant in toolbar

### PR DESCRIPTION
**Pull request by fralongo**
### Description

The support does not include `contentHeader` and embedding `ContentLayout`. We will add it at a later stage if needed

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/visual-refresh-toolbar/skeleton/styles.scss [hunk 1]
```diff
@@ -105,7 +112,9 @@
     /* stylelint-disable plugin/no-unsupported-browser-features */
     &:not(:has(> [data-testid])) {
       inline-size: var(#{custom-props.$toolsWidth});
-      border-inline-start: awsui.$border-divider-section-width solid awsui.$color-border-layout;
+      &.tools-open {
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/skeleton/styles.scss:115]: How is this line related to our bugfix?

Comment (fralongo) [reply src/app-layout/visual-refresh-toolbar/skeleton/styles.scss:115]: Without this change, the border is displayed also when tools are closed. It becomes visible when `headerVariant="high-contrast"`

File: src/app-layout/visual-refresh-toolbar/toolbar/index.tsx [hunk 2]
```diff
@@ -173,6 +175,7 @@ export function AppLayoutToolbarImplementation({
       className={clsx(styles['universal-toolbar'], testutilStyles.toolbar, {
         [testutilStyles['mobile-bar']]: isMobile,
         [styles['toolbar-hidden']]: toolbarHidden,
+        [highContrastHeaderClassName]: headerVariant === 'high-contrast',
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/toolbar/index.tsx:178]: Why did we make this change?
